examples/demo-app/src/cloud-providers/google-drive-provider.js

```diff
@@ -0,0 +1,65 @@
+/* global gapi */
+import GoogleProvider from './google-provider';
+import GoogleDriveIcon from '../components/icons/google-drive-icon';
+
+// Array of API discovery doc URLs for APIs
+const DISCOVERY_DOCS = ["https://www.googleapis.com/discovery/v1/apis/drive/v3/rest"];
+
+const SCOPES = ['https://www.googleapis.com/auth/drive.file']; // Restrict to files from this app
+
+// There is a selection of access scopes for Google Drive, e.g.
+// - https://www.googleapis.com/auth/drive	See, edit, create, and delete all of your Google Drive files
+// - https://www.googleapis.com/auth/drive.appdata	View and manage its own configuration data in your Google Drive
+// - https://www.googleapis.com/auth/drive.file	View and manage Google Drive files and folders that you have opened or created with this app
+// See https://developers.google.com/identity/protocols/googlescopes#drivev3
+
+export default class GoogleDriveProvider extends GoogleProvider {
+  /**
+   * To access Google Drive, specify it in scopes
+   * param scopes = ['https://www.googleapis.com/auth/drive.file'];
+   */
+  constructor(appName, clientId, appKey, scopes = SCOPES) {
```

Comment:
We should try to make to make this consistent across all cloud providers. 
Right now Dropbox provider accepts an icon where here we have different parameters. 

I think we can follow a more flexible pattern:

constructor(appName, clientId, options = {})

options can handle all different requirement for each cloud provider